crates/evm/db.rs

```diff
@@ -44,8 +44,8 @@ impl revm::Database for StoreWrapper {
     fn storage(&mut self, address: RevmAddress, index: RevmU256) -> Result<RevmU256, Self::Error> {
         Ok(self
             .store
-            .get_storage_at(
-                self.block_number,
+            .get_storage_at_hash(
+                self.block_hash,
```

Comment:
There might be more than one block with the same block number but different hash. When you get block by number, you are getting the canonical block. The hash is the unique identifier. This change happens because you want to be able to calculate the state transition on non-canonical blocks too.